Updates for jenkins-workspace binary temporaries repository

joaocgreis opened the pull request:
This does two things:
- Increase MaxSartups in `sshd_config`. This is necessary when many workers running tests try to access the temp repo to download binaries simultaneously. Not sure if there is an optimal value for this. Since the previous value seemed to just barely not be enough, I increased it 10x considering it is possible that more than one job might be started at the same time.
- Weekly clean-up of the repository. This deletes branches where the last commit is more than one month old. This is not ideal, but since https://github.com/nodejs/build/issues/1496 the branches no longer have the number of the job that started them, so looking at Jenkins jobs is no longer possible. This replaces `git-clean-temp-repo`, which was broken not only because of this but also because unauthenticated access to Jenkins is disabled.

refack commented:
Question: any reason not to use Jenkins' built-in cron (just so that we don't have configuration in too many places)?
![image](https://user-images.githubusercontent.com/96947/46481923-724c8380-c7c2-11e8-9e26-0ab43ff7a174.png)

joaocgreis commented:
@refack was there before, I removed it because the job is broken. Deleting the job and archiving as XML here is the next step there.

refack commented on ansible/roles/jenkins-workspace/files/clean_binary_tmp.sh:
```diff
@@ -0,0 +1,16 @@
+  fi
+done
+
+git prune
```
maybe add a ```nohup git gc >& ~binary_tmp/git_gc_`date +"%s"`.log ```?
That way we won't get an auto `gc` in the middle of a future `fetch`

joaocgreis replied:
Please see below about gc.

refack commented on ansible/roles/jenkins-workspace/files/clean_binary_tmp.sh:
```diff
@@ -0,0 +1,16 @@
+git fetch origin +master:master
+
+for b in $(git branch | sed /\*/d); do
+  if [ -z "$(git log -1 --since='4 weeks ago' -s $b)" ]; then
```
```suggested
  if [ -z "$(git log -1 --since='5 days ago' -s $b)" ]; then
```
IMO 5 days is enough. After five days we should run a rebuild not a resume (since `master` probably progressed).

rvagg replied:
agreed, 4 weeks is a long time, any good reason for that @joaocgreis?

mhdawson replied:
I find that after a week the jobs are most often no longer in the history anyway.  Maybe 7 days?

joaocgreis replied:
7 days sounds reasonable, updated.

joaocgreis commented:
Updated (9b6f4b03aa2ff8f00c2c52a4056170a73b36a761), PTAL.

- Changed the expiry time to 7 days.
- Changed the script to run daily.
- Disabled git auto gc, since it can interfere with fetching from test jobs. I don't think there's a need to run the full gc in the script, since repacking does not make sense for this usage and we already do the pruning explicitly. The full `git gc` takes hours to complete, so I've let it out, but we can add it to the script if there's a need for it.
- Changed logging to not include bash output and added another `du` in the beginning.

joaocgreis commented:
Landed in https://github.com/nodejs/build/compare/a1957ef2c75ad576f5932adc2ed32852b4e58df3...5961359b794190575ebc53b304e93d38ef25c552
